### ISSUE TITLE
Mac Flaky test: ModifiedPathsTests.DeletedTempFolderDeletesFilesFromModifiedFiles

### ISSUE THREAD

**Issue by nickgra** [Microsoft/VFSForGit#419]
https://dev.azure.com/mseng/AzureDevOps/_build/results?buildId=7564298&view=logs 

```
2018-10-23T00:48:08.6825990Z 1) Failed : GVFS.FunctionalTests.Tests.EnlistmentPerTestCase.ModifiedPathsTests.DeletedTempFolderDeletesFilesFromModifiedFiles(GVFS.FunctionalTests.FileSystemRunners.SystemIORunner)
2018-10-23T00:48:08.6826080Z Unexpected matching entry found in {A Temp/,A .gitattributes,}
2018-10-23T00:48:08.6826150Z Expected: null
2018-10-23T00:48:08.6826210Z But was: "A Temp/"
2018-10-23T00:48:08.6826320Z at GVFS.Tests.Should.ValueShouldExtensions.ShouldEqual[T](T actualValue, T expectedValue, String message) in /Users/gvfsadmin/agent/_work/4/s/GVFS/GVFS.Tests/Should/ValueShouldExtensions.cs:line 34
2018-10-23T00:48:08.6826470Z at GVFS.Tests.Should.EnumerableShouldExtensions.ShouldNotContain[T](IEnumerable`1 group, Func`2 predicate) in /Users/gvfsadmin/agent/_work/4/s/GVFS/GVFS.Tests/Should/EnumerableShouldExtensions.cs:line 51
2018-10-23T00:48:08.6826620Z at GVFS.FunctionalTests.Tools.GVFSHelpers.ModifiedPathsShouldNotContain(GVFSFunctionalTestEnlistment enlistment, FileSystemRunner fileSystem, String[] gitPaths) in /Users/gvfsadmin/agent/_work/4/s/GVFS/GVFS.FunctionalTests/Tools/GVFSHelpers.cs:line 127
2018-10-23T00:48:08.6826790Z at GVFS.FunctionalTests.Tests.EnlistmentPerTestCase.ModifiedPathsTests.DeletedTempFolderDeletesFilesFromModifiedFiles(FileSystemRunner fileSystem) in /Users/gvfsadmin/agent/_work/4/s/GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerTestCase/ModifiedPathsTests.cs:line 74
2018-10-23T00:48:08.6827060Z
```

**Comment by nickgra** [comment]
Another hit:
https://dev.azure.com/mseng/AzureDevOps/_build/results?buildId=7573103&view=logs

```
2018-10-23T23:45:21.8085830Z 
2018-10-23T23:45:21.8085920Z 1) Failed : GVFS.FunctionalTests.Tests.EnlistmentPerTestCase.ModifiedPathsTests.DeletedTempFileIsRemovedFromModifiedFiles(GVFS.FunctionalTests.FileSystemRunners.SystemIORunner)
2018-10-23T23:45:21.8086220Z   Unexpected matching entry found in {A .gitattributes,A temp.txt,}
2018-10-23T23:45:21.8086300Z   Expected: null
2018-10-23T23:45:21.8086360Z   But was:  "A temp.txt"
2018-10-23T23:45:21.8086470Z    at GVFS.Tests.Should.ValueShouldExtensions.ShouldEqual[T](T actualValue, T expectedValue, String message) in /Users/gvfsadmin/agent/_work/4/s/GVFS/GVFS.Tests/Should/ValueShouldExtensions.cs:line 34
2018-10-23T23:45:21.8086620Z    at GVFS.Tests.Should.EnumerableShouldExtensions.ShouldNotContain[T](IEnumerable`1 group, Func`2 predicate) in /Users/gvfsadmin/agent/_work/4/s/GVFS/GVFS.Tests/Should/EnumerableShouldExtensions.cs:line 51
2018-10-23T23:45:21.8086790Z    at GVFS.FunctionalTests.Tools.GVFSHelpers.ModifiedPathsShouldNotContain(GVFSFunctionalTestEnlistment enlistment, FileSystemRunner fileSystem, String[] gitPaths) in /Users/gvfsadmin/agent/_work/4/s/GVFS/GVFS.FunctionalTests/Tools/GVFSHelpers.cs:line 127
2018-10-23T23:45:21.8086970Z    at GVFS.FunctionalTests.Tests.EnlistmentPerTestCase.ModifiedPathsTests.DeletedTempFileIsRemovedFromModifiedFiles(FileSystemRunner fileSystem) in /Users/gvfsadmin/agent/_work/4/s/GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerTestCase/ModifiedPathsTests.cs:line 50
2018-10-23T23:45:21.8087070Z ```

**Comment by kewillford** [comment]
This should be fixed with #401
